src/core/manager.c

```diff
@@ -3459,12 +3459,12 @@ void manager_override_watchdog(Manager *m, WatchdogType t, usec_t timeout) {
         if (m->watchdog_overridden[t] == timeout)
                 return;
 
-        if (t == WATCHDOG_RUNTIME) {
-                usec_t usec = timestamp_is_set(timeout) ? timeout : m->watchdog[t];
-
-                (void) watchdog_setup(usec);
-        } else if (t == WATCHDOG_PRETIMEOUT)
-                (void) watchdog_setup_pretimeout(timeout);
+        if (timeout != USEC_INFINITY) {
```

Comment:
doesn't this change the behaviour? previously if the timeout was infinity, watchdog_setup was called with m->watchdog[t], now it's not called at all, was that intentional, and if so, why?

Reply:
I think I was thinking that USEC_INFINITY is a nop, so we can skip calling watchdog_setup[_pretimeout]. But USEC_INFINITY here means "clear the override and use the original timeout value" right? In that case, yes we should be doing that.